libqaul/src/api/models.rs

```diff
@@ -28,6 +27,16 @@ pub enum UserAuth {
     Trusted(Identity, String),
 }
 
+impl UserAuth {
+    /// Returns an error if the UserAuth isn't Trusted.
+    pub fn require_trusted(self) -> QaulResult<(Identity, String)> {
```

Comment:
That's a nice util function! :+1: 

Although maybe make it `pub(crate)` and shorten the name to something like `trusted` ? (just as an idea)

Reply:
Yeah, good call.